hopsanc/src/hopsanc.cpp

```diff
@@ -0,0 +1,282 @@
+
+    printWaitingMessages(gHopsanCore, false, false);
+    return 0;
+}
+
+int getTimeVector(double *data)
+{
+    if(!spCoreComponentSystem) {
+        std::cout << "Error: No model is loaded.\n";
+        return -1;
+    }
+    memcpy(data,spCoreComponentSystem->getLogTimeVector()->data(), spCoreComponentSystem->getNumActuallyLoggedSamples()*sizeof(double));
+    return 0;
+}
+
+
+
+int setParameter(const char *name, const char *value)
+{
+    if(!spCoreComponentSystem) {
+        std::cout << "Error: No model is loaded.\n";
+        return -1;
+    }
+
+    //Parse arguments
+    hopsan::HString nameStr(name);
+    hopsan::HVector<hopsan::HString> sysVec = nameStr.split('|');
+    hopsan::HVector<hopsan::HString> nameVec = sysVec.last().split('.');
+    sysVec.resize(sysVec.size()-1);
+
+    //Generate component name and parameter name
+    hopsan::HString compName, parName;
+    if(nameVec.size() == 1) {   //System parameter
+        compName = "";
+        parName = nameVec[0];
+    }
+    else if(nameVec.size() == 2) { //Constant
+        compName = nameVec[0];
+        parName = nameVec[1];
+    }
+    else if(nameVec.size() == 3) { //Input variable
+        compName = nameVec[0];
+        parName = nameVec[1]+"#"+nameVec[2];
+    }
+    else {
+        std::cout << "Error: Parameter name not specified.\n";
+        return -1;
+    }
+
+    //Find system
+    hopsan::ComponentSystem *pSystem = spCoreComponentSystem;
+    for(size_t i=0; i<sysVec.size(); ++i) {
+        pSystem = pSystem->getSubComponentSystem(sysVec[i]);
+        if(!pSystem) {
+            std::cout << "Error: Subsystem not found: " << sysVec[i].c_str() << "\n";
+            return -1;
+        }
+    }
+
+    if(compName.empty()) {   //Set system parameter
+        if(pSystem->setParameterValue(parName, hopsan::HString(value))) {
+            return 0;
+        }
+        else {
+            std::cout << "Error: Failed to set parameter value: " << parName.c_str() << "\n";
+            return -1;
+        }
+    }
+    else if(nameVec.size() == 2 || nameVec.size() == 3) { //Set constant or input variable
+        hopsan::Component *pComp = pSystem->getSubComponent(compName);
+        if(!pComp) {
+            std::cout << "Error: No such component: " << compName.c_str() << "\n";
+            return -1;
+        }
+        if(pComp->setParameterValue(parName, hopsan::HString(value))) {
+            return 0;
+        }
+        else {
+            std::cout << "Error: Failed to set parameter value: " << parName.c_str() << "\n";
+            return -1;
+        }
+    }
+
+    std::cout << "Error: Wrong number of arguments.\n";
+    return -1;
+}
+
+int setLogSamples(unsigned long value)
+{
+    if(!spCoreComponentSystem) {
+        std::cout << "Error: No model is loaded.\n";
+        return -1;
+    }
+    std::cout << "Setting samples to " << size_t(value) << "\n";
+    spCoreComponentSystem->setNumLogSamples(value);
+    return 0;
+}
+
+size_t getLogSamples()
```

Comment:
size_t is not a C type I think, (maybe i newer C standards)

Reply:
It was defined in C99, so I think it is safe to use it. But I will change both functions so they have the same type.